src/sealock.c

```diff
@@ -399,19 +399,40 @@ void sealock_get_active_layers(sealock_state_t *lock) {
 }
 
 // Collect aggregate of quantity from dfm layer buffer into scalar.
-static double sealock_collect(dfm_volumes_t *volumes, double* buffer_ptr) { 
+static double sealock_collect(profile_t *profile, dfm_volumes_t *volumes, double* buffer_ptr) { 
   double aggregate = 0.0;
 
   assert(volumes);
   assert(buffer_ptr);
 
   int first = volumes->first_active_cell;
   int last = first + volumes->num_active_cells - 1;
+  double previous_volume = 0.0;
+  double next_volume = 0.0;
+  double used_volume = 0.0;
+  double used_fraction = 0.0;
 
   for (int i = first; i <= last; i++) {
-    aggregate += buffer_ptr[i]; // TODO: Check if we need * volumes->volumes[i] or normalized[i] here?
+    next_volume += volumes->normalized[i];
+    double fraction = -1.0 * integrate_piecewise_linear_profile(profile, previous_volume, next_volume);
+    log_debug("Layer %d, value = %g, volume = %g, profile fraction = %g\n", i, buffer_ptr[i], volumes->normalized[i], fraction);
+    if (fraction > 0) {
```

Comment:
I need more comments or some explanation here about the sign of fraction. 
Why is it multiplied with -1.0 and why is it only added if fraction > 0> That implies, you only add the pieces where integrate_piecewise_linear_profile yields a negative volume?

Reply:
I added some explanatory comments.